src/clients/BundleDataClient/utils/SuperstructUtils.ts

```diff
@@ -54,6 +56,7 @@ const SortableEventSS = {
 };
 
 const V3DepositSS = {
+  messageHash: defaulted(string(), UNDEFINED_MESSAGE_HASH),
```

Comment:
`defaulted()` will establish an instance of `messageHash` with the value `""`. I looked at superstruct for ways of dynamically assigning this (i.e. computing on the fly based on the rest of the object). It kinda seems like superstruct can support this, but we have a lot of nested structures in our object validations and I lost patience with it.

In any case, this is backwards compatible because we can populate messageHash values for existing deposits that were not posted with the field. I'd also consider dropping it from data posted to arweave, since there is some odd redundancy between `message` and `messageHash`, but we don't expect to verify one from the other when we pull them down simultaneously.